cads_adaptors/tools/url_tools.py

```diff
@@ -35,21 +36,20 @@ def try_download(urls: List[str]) -> List[str]:
     for url in urls:
         path = urllib.parse.urlparse(url).path.lstrip("/")
         dir = os.path.dirname(path)
-        os.makedirs(dir, exist_ok=True)
+        if dir:
+            os.makedirs(dir, exist_ok=True)
         try:
             multiurl.download(url, path)
+        except Exception as exc:
+            excs.append({url: exc})
```

Comment:
This list was populated to show all the errors in the RuntimeError message, but that feature is removed in this PR.
I think you should either restore that feature, or get rid of the `excs` list as well.

Reply:
This was intentional to try and make the error messages received by users more undertandable, but you're right, I no longer need the excs list as we now dump those to internal logs straight away